.github/workflows/PR-CI.yml

```diff
@@ -0,0 +1,47 @@
+name: PR-CI
+
+on:
+    pull_request:
+        branches: '**'
+
+jobs:
+    install-dependencies:
+        runs-on: ubuntu-latest
+        container: node:21-alpine
+
+        steps:
+            - uses: actions/checkout@v4
+
+            - name: install dependencies
+              run: yarn install --frozen-lockfile
+
+            - name: list files
+              run: ls -la /
+
+            - name: Tar node_modules directory
+              run: tar czf node_modules.tar.gz -C node_modules .
+
+            - name: Upload node_modules_artifact
+              uses: actions/upload-artifact@v4
+              with:
+                  name: node_modules
+                  path: node_modules.tar.gz
+
+    build:
```

Comment:
perhaps you could also add jobs to test prettier and eslint before the build job